packages/cli-kit/src/output.ts

```diff
@@ -347,9 +352,9 @@ export const completed = (content: Message) => {
  * Note: Debug messages are sent through the standard output.
  * @param content {string} The content to be output to the user.
  */
-export const debug = (content: Message) => {
+export const debug = (content: Message, sync = false) => {
```

Comment:
As a user of a logging API, I wouldn't expect to be presented with a synchronous/asynchronous decision. If we are concerned about logs being blocking because we are doing expensive IO (e.g. writing to a log file), I'd revisit the logging approach to off-load the work from the main process. Node has a [building block for that](https://nodejs.org/api/worker_threads.html that libraries like [Pino](https://github.com/pinojs/pino) use under the hood.